File: Python3/283.Move_Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        zero_counter = 0
        i = 0
        
        while i + zero_counter < n:
            if nums[i + zero_counter] == 0:
                zero_counter += 1
            else:
                if zero_counter > 0:
                    nums[i] = nums[i + zero_counter]
                i += 1
        
        while i < n:
            nums[i] = 0
            i += 1

File: Java/283.Move_Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length;
        int i = 0, zero_count = 0;
        while (i + zero_count < n) {
            if (nums[i + zero_count] == 0) ++zero_count;
            else {
                nums[i] = nums[i + zero_count];
                ++i;
            }
        }
        while (i < n) {
            nums[i++] = 0;
        }
    }
}